src/Microsoft.DotNet.Build.Tasks/IsRestoreRequired.cs

```diff
@@ -0,0 +1,171 @@
+        {
+            // match nuget behavior
+            return Path.GetFileName(Path.GetDirectoryName(projectJsonPath));
+        }
+
+        // Lightweight lock file parser that only reads the package identities
+        private static IEnumerable<PackageIdentity> ReadLibraries(string lockFilePath)
+        {
+            using (var stream = File.OpenRead(lockFilePath))
+            using (var textReader = new StreamReader(stream))
+            using (var jsonReader = new JsonTextReader(textReader))
+            {
+                while (jsonReader.TokenType != JsonToken.StartObject)
+                {
+                    EnsureRead(jsonReader);
+                }
+
+                ReadToValue(jsonReader, JsonToken.PropertyName, "libraries");
+                EnsureRead(jsonReader); // StartObject
+                EnsureRead(jsonReader); // Property name
+
+                while (jsonReader.TokenType != JsonToken.EndObject)
+                {
+                    var packageString = jsonReader.Value as string;
+
+                    if (packageString == null)
+                    {
+                        throw new InvalidDataException($"Unexpected entry in {lockFilePath} at line {jsonReader.LineNumber}, position {jsonReader.LinePosition}.  Expected property name.");
+                    }
+
+                    var parts = packageString.Split('/');
+                    if (parts.Length != 2)
+                    {
+                        throw new InvalidDataException($"Unexpected entry in {lockFilePath} at line {jsonReader.LineNumber}, position {jsonReader.LinePosition}.  Expected string of format id/version.");
+                    }
+
+                    EnsureRead(jsonReader); // StartObject
+                    ReadToValue(jsonReader, JsonToken.PropertyName, "type");
+                    EnsureRead(jsonReader); // value
+                    if (jsonReader.Value.Equals("package"))
+                    {
+                        yield return new PackageIdentity(parts[0], NuGetVersion.Parse(parts[1]));
+                    }
+
+                    ReadToToken(jsonReader, JsonToken.EndObject); // Move to End
+                    EnsureRead(jsonReader); // next PropertyName
+                }
+            }
+        }
+
+        private static void ReadToValue(JsonTextReader jsonReader, JsonToken tokenType, object value)
+        {
+            while (jsonReader.Value == null || jsonReader.TokenType != tokenType || !jsonReader.Value.Equals(value))
+            {
+                EnsureRead(jsonReader);
+            }
+        }
+        private static void ReadToToken(JsonTextReader jsonReader, JsonToken tokenType)
+        {
+            while (jsonReader.TokenType != tokenType)
+            {
+                EnsureRead(jsonReader);
+            }
+        }
+
+        private static void EnsureRead(JsonTextReader reader)
+        {
+            if (!reader.Read())
+            {
+                throw new EndOfStreamException();
```

Comment:
We never catch it.  We never expect to hit it.  This would mean that the lock file format didn't meet our expectations.  I considered adding more checks to have a better error (eg: catch when the depth changed as we were advancing the reader) but I decided instead to keep it simple.  Ideally the next major nuget update will let us delete this code.
the task